function/pkg/client/status.go

```diff
@@ -0,0 +1,110 @@
+package client
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type StatusType int
+
+func (t StatusType) String() string {
+	switch t {
+	case StatusTypeDeleted:
+		return "deleted"
+	case StatusTypeSkipped:
+		return "skipped"
+	case StatusTypeFailed:
+		return "failed"
+	case StatusTypeCreated:
+		return "created"
+	case StatusTypeUpdated:
+		return "updated"
+	default:
+		return "unknown"
+	}
+}
+
+type Result = interface{}
```

Comment:
Shouldn't this be just `type Result interface{}`? Is the `=` doing something special? Never seen this construct before.

Reply:
It's an alias; it's a leftover; we will remove it; there is a nice article on aliases on medium [here](https://medium.com/a-journey-with-go/go-aliases-simple-and-efficient-8506d93b079e)

Reply:
Yes I remembered while reviewing the other PR.
My understanding is that aliases should only be used as in between solution for codebase refactoring if needed:

See: https://talks.golang.org/2016/refactor.article